lib/at_monitor/at_monitor.c

```diff
@@ -83,7 +94,9 @@ void at_monitor_dispatch(const char *notif)
 	strcpy(at_notif->data, notif);
 
 	k_fifo_put(&at_monitor_fifo, at_notif);
-	k_work_submit(&at_monitor_work);
+
+	/* Dispatch to queue */
+	k_work_submit_to_queue(&WORKQUEUE, &at_monitor_work);
```

Comment:
I think this is simpler to read (with corresponding changes to declarations):
```suggestion
	/* Dispatch to queue */
#if CONFIG_AT_MONITOR_USE_DEDICATED_WORKQUEUE
	k_work_submit_to_queue(&at_monitor_work_q, &at_monitor_work);
#else
	k_work_submit(&at_monitor_work);
#endif
```

Reply:
ok, updated